be/src/runtime/file_result_writer.cpp

```diff
@@ -131,12 +132,14 @@ Status FileResultWriter::_create_next_file_writer() {
 Status FileResultWriter::_create_file_writer(const std::string& file_name) {
     if (_storage_type == TStorageBackendType::LOCAL) {
         _file_writer = new LocalFileWriter(file_name, 0 /* start offset */);
-    } else if (_storage_type == TStorageBackendType::BROKER){
+    } else if (_storage_type == TStorageBackendType::BROKER) {
         _file_writer =
                 new BrokerWriter(_state->exec_env(), _file_opts->broker_addresses,
                                  _file_opts->broker_properties, file_name, 0 /*start offset*/);
     } else if (_storage_type == TStorageBackendType::S3) {
         _file_writer =  new S3Writer(_file_opts->broker_properties, file_name, 0 /* offset */);
+    } else if (_storage_type == TStorageBackendType::HDFS) {
+        _file_writer = new HDFSWriter(const_cast<std::map<std::string, std::string>&>(_file_opts->broker_properties), file_name);
```

Comment:
Why need a `const_cast` here but `S3Writer` don't?

Reply:
The HDFS writer needs to parse properties so here is a ```const_cast```